### PR TITLE
HexEditWidget and NodeTreeWidget merged into single tab.

### PR DESCRIPTION
**Pull request by jacek-izykowski**
Features related to interactions between minimap and hex view
are out of the scope of this commit.

### CODE REVIEW

**Comment by altran01** [comment]
Would it also be possible to remember which sidebars are open when moving hex-view tab around? Currently all tabs are automatically closed when you move tab in any way (apart from reordering tabs).

File: src/ui/hexedit.cc [hunk 1]
```diff
@@ -413,6 +413,7 @@ void HexEdit::getRangeFromIndex(QModelIndex index, qint64 *start,
 void HexEdit::paintEvent(QPaintEvent *event) {
   QPainter painter(viewport());
 
+#if 0
```
Comment (altran01) [inline src/ui/hexedit.cc]: At least add some comment as to why is this part of code permanently disabled?

File: src/ui/hexeditwidget.cc [hunk 2]
```diff
@@ -137,8 +150,19 @@ void HexEditWidget::createActions() {
...
-  connect(show_node_tree_act_, SIGNAL(triggered()),
-      this, SLOT(showNodeTree()));
+  show_node_tree_act_->setCheckable(true);
+  show_node_tree_act_->setChecked(true);
+  connect(show_node_tree_act_, SIGNAL(toggled(bool)),
+      this, SIGNAL(showNodeTree(bool)));
+
+  show_minimap_act_ = new QAction(QIcon(":/images/show_minimap.png"),
+      tr("&Minimap"), this);
+  show_minimap_act_->setToolTip(tr("Node tree"));
```
Comment (altran01) [inline src/ui/hexeditwidget.cc]: copy-paste error :)

File: src/ui/veles_mainwindow.cc [hunk 3]
```diff
@@ -297,6 +298,7 @@ void VelesMainWindow::createHexEditTab(QString fileName,
   QSharedPointer<QItemSelectionModel> selection_model(
       new QItemSelectionModel(data_model.data()));
 
+#if 0
```
Comment (altran01) [inline src/ui/veles_mainwindow.cc]: Again some comment here ?

**Comment by altran01** [comment]
Another bug I've noticed is that changing hex-edit columns number in options doesn't have an effect on already open tabs anymore - it is only applied to newly opened ones.

**Comment by jacek-izykowski** [comment]
@altran01 Ok. But it seems to be out of the scope of this pull request.

File: src/ui/hexeditwidget.cc [hunk 4]
```diff
@@ -265,5 +332,22 @@ void HexEditWidget::newBinData() {
   visualisation_act_->setEnabled(data_model_->binData().size() > 0);
 }
 
+void HexEditWidget::selectionChanged(qint64 start_addr,
+    qint64 selection_size) {
+  selection_label_->setText(
```
Comment (altran01) [inline src/ui/hexeditwidget.cc]: I think that one of assumptions we made about this label was that user should be able to easily select it and copy its content (this wasn't true for previous version as well, however we might want to change it now).

File: src/ui/hexedit.cc [hunk 5]
```diff
@@ -434,6 +435,7 @@ void HexEdit::paintEvent(QPaintEvent *event) {
   painter.drawText(startMargin_ - startPosX_,
                    separatorLength - horizontalAreaSpaceWidth_,
                    statusBarText());
```
Comment (altran01) [inline src/ui/hexedit.cc]: Overall I'm not a big fan of leaving unused code like this lying around - if someone would like to use it again later on then he could simply get it from git history. This introduces some unnecessary complications - like unused methods i.e. statusBarText - without checking that it now isn't called anywhere one could be surprised that editing it doesn't have any effect.

File: src/ui/hexedit.cc [hunk 6]
```diff
@@ -434,6 +435,7 @@ void HexEdit::paintEvent(QPaintEvent *event) {
...
+#endif
 
   for (auto rowNum = startRow_;
```
Comment (altran01) [inline src/ui/hexedit.cc]: After removing the code above, we now have an quite big empty area at bottom of our hex-view - maybe we should do something about it ?

File: src/ui/hexeditwidget.cc (hunk 4 above)
Comment (jacek-izykowski) [inline src/ui/hexeditwidget.cc]: Fixed.

**Comment by altran01** [comment]
Please rebase.

**Pull request merged**